app/schema/widgets/currency_widget.py

```diff
@@ -0,0 +1,30 @@
+from app.schema.widget import Widget
+from flask import render_template
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class CurrencyWidget(Widget):
+    def __init__(self, name):
+        super().__init__(name)
+
+    def render(self, schema, state):
+        widget_params = {
+            'answer': {
```

Comment:
Following on from this, the internal structures of the widget is subject to being refactored somewhat once we have some additional widget types to play with.  The next step would be to add an "Other" option to the radio group and checkbox group widgets allowing user-specified options.

Hamish is planning to continue tidying up the template layer which will also drive refactoring opportunities within widgets.